benchmarks/yamauchi_takei_2016_anelasticity/anelasticity_temperature.cc

```diff
@@ -0,0 +1,408 @@
+/*
+  Copyright (C) 2011 - 2018 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/global.h>
+#include "anelasticity_temperature.h"
+#include <aspect/material_model/interface.h>
+#include <aspect/initial_composition/interface.h>
+#include <aspect/adiabatic_conditions/interface.h>
+#include <aspect/simulator_access.h>
+#include <cmath>
+#include <algorithm>
+#include <functional>
+
+#include <boost/lexical_cast.hpp>
+#include <boost/math/tools/minima.hpp>
+
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    template <int dim>
+    AnelasticVs2T<dim>::AnelasticVs2T ()
+    {}
+
+    template <int dim>
+    void
+    AnelasticVs2T<dim>::initialize ()
+    {
+      Utilities::AsciiDataInitial<dim>::initialize(1);
+    }
+
+// set up Vs function that Brent minimization operates on
+    template <int dim>
+    double
+    AnelasticVs2T<dim>::
+    fVs (double x,
+         const double      depth,
+         const double      absolute_Vs,
+         const double      mu0,
+         const double      dmudT,
+         const double      dmudP,
+         const double      viscosity_prefactor,
+         const double      activation_energy,
+         const double      activation_volume,
+         const double      solidus_gradient,
+         const bool      use_original_model) const
+    {
+      return abs(yamauchi_takei_Vs(x,depth,mu0,dmudT,dmudP,viscosity_prefactor,activation_energy,activation_volume
```

Comment:
You'll have to write this as `std::abs` for complicated and historical reasons :-( Maybe the same in other places below -- please do a search-replace.